src/charm.py

```diff
@@ -41,83 +70,51 @@ def __init__(self, *args):
 
     def _update_prometheus_jobs(self, _):
         """Setup Prometheus scrape configuration for external targets."""
-        self.unit.status = ActiveStatus()
         if not self.unit.is_leader():
             return
 
-        jobs = self._scrape_jobs()
-        if not jobs:
-            return
+        if jobs := self._scrape_jobs():
+            for relation in self.model.relations[self._prometheus_relation]:
+                relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
 
-        for relation in self.model.relations[self._prometheus_relation]:
-            relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
+            self.unit.status = ActiveStatus()
 
-    def _scrape_jobs(self):
-        targets = self._targets()
+    def _scrape_jobs(self) -> list:
+        if targets := self._targets():
+            static_config = {"targets": targets}
+            if labels := self._labels():
+                static_config.update(labels=labels)
 
-        jobs = (
-            [
+            return [
                 {
                     "job_name": self._job_name(),
                     "metrics_path": self.model.config["metrics-path"],
```

Comment:
Prometheus has already got a default for the path:
```
# The HTTP resource path on which to fetch metrics from targets.
[ metrics_path: <path> | default = /metrics ]
```
So maybe we should be duplicating it (hardcoding it ourselves)

Reply:
I am in favor of removing it if left default